src/main/c/netty_quic_quiche.c

```diff
@@ -347,6 +356,27 @@ static void netty_quiche_conn_free(JNIEnv* env, jclass clazz, jlong conn) {
     quiche_conn_free((quiche_conn *) conn);
 }
 
+static jobjectArray netty_quiche_conn_peer_error0(JNIEnv* env, jclass clazz, jlong conn) {
+    bool is_app = false;
+    uint64_t error_code = 0;
+    const uint8_t *reason = NULL;
+    size_t reason_len = 0;
+
+    bool peer_error = quiche_conn_peer_error((quiche_conn *) conn,
+                            &is_app,
+                            &error_code,
+                            &reason,
+                            &reason_len);
+    if (peer_error) {
+        jobjectArray array = (*env)->NewObjectArray(env, 3, object_class, NULL);
+        (*env)->SetObjectArrayElement(env, array, 0, (*env)->CallStaticObjectMethod(env, boolean_class, boolean_class_valueof, is_app ? JNI_TRUE : JNI_FALSE));
+        (*env)->SetObjectArrayElement(env, array, 1, (*env)->NewObject(env, integer_class, integer_class_init, (jint) error_code));
+        (*env)->SetObjectArrayElement(env, array, 2, to_byte_array(env, reason, reason_len));
+        return array;
```

Comment:
Why this array based transport?

Reply:
You mean why return an array and not just our "class directly"? Usually I prefer to return "standard" objects from JNI and do the transformation in the java layer. This makes things easier also when it comes to shading etc.